src/components/button/button.test.tsx

```diff
@@ -1,12 +1,63 @@
 import React from 'react';
-import { render, screen } from '@testing-library/react';
+import { render, screen, fireEvent } from '@testing-library/react';
+import '@testing-library/jest-dom';
 
 import Button from './button';
+import { ButtonProps } from './button';
+
+const clickEvent = jest.fn();
+const defaultButton: ButtonProps = {
+  label: 'Button',
+  onClick: clickEvent,
+};
+
+function sut(
+  props: ButtonProps = defaultButton
+): HTMLElement {
+  render(<Button {...props} />);
+  return screen.getByRole('button', { name: props.label });
+}
 
 describe('Button', () => {
-  test('renders the Button component', () => {
+  test('should render the Button component', () => {
     const label = 'Hello world!';
     render(<Button label={label} />);
     expect(screen.queryAllByText(label)).toHaveLength(1);
   });
+
+  test('should execute event when the button is clicked', () => {
+    fireEvent.click(sut({  ...defaultButton }));
```

Comment:
```suggestion
    userEvent.click(sut({  ...defaultButton }));
```

change to userEvent, fireEvent not is recommended.